Source/WebKit/NetworkProcess/webtransport/cocoa/NetworkTransportSessionCocoa.mm

```diff
@@ -47,20 +50,74 @@
 }
 
 #if HAVE(WEB_TRANSPORT)
-static RetainPtr<nw_parameters_t> createParameters(WebCore::SecurityOriginData&& origin)
+
+static void didReceiveServerTrustChallenge(Ref<NetworkConnectionToWebProcess>&& connectionToWebProcess, URL&& url, WebKit::WebPageProxyIdentifier&& pageID, struct WebCore::ClientOrigin&& clientOrigin, sec_trust_t trust, sec_protocol_verify_complete_t completion)
+{
+    uint16_t port = url.port() ? *url.port() : *defaultPortForProtocol(url.protocol());
+    RetainPtr secTrust = adoptCF(sec_trust_copy_ref(trust));
+    RetainPtr protectionSpace = adoptNS([[NSURLProtectionSpace alloc] initWithHost:url.host().createNSString().get() port:port protocol:NSURLProtectionSpaceHTTPS realm:nil authenticationMethod:NSURLAuthenticationMethodServerTrust]);
+    [protectionSpace _setServerTrust:secTrust.get()];
+
+    id<NSURLAuthenticationChallengeSender> sender = nil;
+    RetainPtr challenge = adoptNS([[NSURLAuthenticationChallenge alloc] initWithProtectionSpace:protectionSpace.get() proposedCredential:nil previousFailureCount:0 failureResponse:nil error:nil sender:sender]);
+
+    auto challengeCompletionHandler = [completion = makeBlockPtr(completion), secTrust = WTFMove(secTrust)] (AuthenticationChallengeDisposition disposition, const WebCore::Credential& credential) {
+        switch (disposition) {
+        case AuthenticationChallengeDisposition::UseCredential: {
+            if (!credential.isEmpty())
+                completion(true);
+        }
+        FALLTHROUGH;
+        case AuthenticationChallengeDisposition::RejectProtectionSpaceAndContinue:
+        case AuthenticationChallengeDisposition::PerformDefaultHandling: {
+            OSStatus status = SecTrustEvaluateAsyncWithError(secTrust.get(), dispatch_get_main_queue(), makeBlockPtr([completion = completion](SecTrustRef trustRef, bool result, CFErrorRef error) {
+                completion(result);
+            }).get());
+            if (status != errSecSuccess)
+                completion(false);
+            return;
+        }
+        case AuthenticationChallengeDisposition::Cancel:
+            completion(false);
+            return;
+        }
+        RELEASE_ASSERT_NOT_REACHED();
+    };
+
+    auto* sessionCocoa = static_cast<NetworkSessionCocoa*>(connectionToWebProcess->networkProcess().networkSession(connectionToWebProcess->sessionID()));
+
+    if (sessionCocoa && sessionCocoa->fastServerTrustEvaluationEnabled()) {
+        auto decisionHandler = makeBlockPtr([sessionCocoa = WeakPtr { *sessionCocoa }, connectionToWebProcess = WTFMove(connectionToWebProcess), pageID = WTFMove(pageID), clientOrigin = WTFMove(clientOrigin), challengeCompletionHandler = WTFMove(challengeCompletionHandler)](NSURLAuthenticationChallenge *challenge, OSStatus trustResult) mutable {
+            auto* session = sessionCocoa.get();
```

Comment:
We don't need to capture the session or check if it's still alive.  We only need to find it to check fastServerTrustEvaluationEnabled then we never need it again in this flow.